bookkeeper_smogon.py

```diff
@@ -33,23 +35,27 @@ def report_reward(self, reward):
         self.reward_sum += reward      # Recall that we must see the outcome of the action
         self.rewards.append(reward)    # before we write down the reward for taking it
     def construct_observation_handler(self):
-        # First, let the observation be the health of both team's Pokemon and also which Pokemon is active.
-        self.state = np.array([0] * 1630)
-        self.shape[-1] = 100
-        self.shape[-2] = 100
-        self.shape[-3] = 100
-        self.shape[-4] = 100
-        self.shape[-5] = 100
-        self.shape[-6] = 100
-        self.shape[-11] = 100
+        self.state = np.array([0] * n)
+        self.state[NUM_POKEMON*2 + TEAM_SIZE + 5] = 100
+        self.state[NUM_POKEMON*2 + TEAM_SIZE + 4] = 100
+        self.state[NUM_POKEMON*2 + TEAM_SIZE + 3] = 100
+        self.state[NUM_POKEMON*2 + TEAM_SIZE + 2] = 100
+        self.state[NUM_POKEMON*2 + TEAM_SIZE + 1] = 100
+        self.state[NUM_POKEMON*2 + TEAM_SIZE] = 100
+        self.state[NUM_POKEMON*2] = 100
+        self.state[NUM_POKEMON*2 + 1] = 100
         # Representing the vector as a matrix makes life easier.
-        self.state.shape = (809*2+12,1)
+        self.state.shape = (n,1)
         self.switch_indices = [0,1,2,3,4,5]
         def report_observation(observation):
             state_updates, self.switch_indices = preprocess_observation(observation)
             for update in state_updates:
                 print(update)
```

Comment:
Make a line of the sort:

    quantity1, quantity2 = update

where quantity1 and quantity2 are meaningful variable names that help document what each element of the update tuple represents.

Reply:
Done